### PR TITLE
Coverity fix

### PR DESCRIPTION
**Pull request by AShivangi**
introduced by https://github.com/Xilinx/XRT/pull/4625

### CODE REVIEW

File: src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp [hunk 1]
```diff
@@ -111,25 +111,20 @@ ReportPlatform::getPropertyTree20202( const xrt_core::device * device,
...
+    logic_uuids.erase(
+      std::remove_if(logic_uuids.begin(), logic_uuids.end(),	
+                      [](const std::string& s) { return s.empty(); }));
   } catch (...) {}
   try {
     interface_uuids = xrt_core::device_query<xrt_core::query::interface_uuids>(device);
-    remove_empty_strings(interface_uuids);
+    interface_uuids.erase(
+      std::remove_if(interface_uuids.begin(), interface_uuids.end(),	
+                  [](const std::string& s) { return s.empty(); }));
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp:127]: This erases one element only and the other moved elements are bogus.  Make sure to add the end() iterator also. See https://en.cppreference.com/w/cpp/container/vector/erase.

std::remove_if returns an iterator to the new logic end of the vector, erase should erase the range [itr, end[ where itr is the new logical end returned by remove_if and end is the end() of the vector itself.
`vec.erase(std::remove_if(b,e,cond), vec.end());`

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp]: Oh, that is what was missing from my initial implementation. Thanks for the clarification. I couldn't figure out why my logic wasn't working earlier. It was missing the end() :)

**Review by stsoe — APPROVED** [review]

**Comment by AShivangi** [comment]
retest this please

**Comment by maxzhen** [comment]
this is xbmgmt2 change. should have no impact on u30.

**Pull request merged**